src/components/demos/read/Demo.js

```diff
@@ -191,12 +190,16 @@ export default function ReadDemo(props) {
             </DemoText>
           )}
         </Box>
-        {readDemo.renderModal &&
-          Modal({
-            update,
-            readDemo,
-            ProgressBar: (props) => <ProgressBar {...props} />,
-          })}
+        {state.renderModal && (
+          <Modal
+            type={'readDemo'}
+            update={update}
+            {...{
+              ...state,
+              ProgressBar: (props) => <ProgressBar {...props} />,
+            }}
```

Comment:
The syntactic gymnastics going on here are ... impressive. Perhaps consider just passing the properties we need or pass `state` as a `state` prop? I don't know what `state` contains so this has some concerns

Reply:
will rename to **demoState**. Would that help?

Reply:
Thought it makes sense, this we are passing a component as prop that takes props. We are using just two, update and readDemo which I've excluded using spread operator